frontend/src/utils/completed-courses-helpers.ts

```diff
@@ -84,83 +69,62 @@ export function getNextTerm(is_summer: boolean, classes: DNDScheduleCourse[]) {
   return classes.splice(0, counter);
 }
 
-export function getSimplifiedCourseData(
-  courses: ISimplifiedCourseData[],
-  completion: string,
-  semester: string = ""
-): ISimplifiedCourseDataAPI[] {
-  return courses.map(course => {
-    return {
-      subject: course.subject,
-      course_id: course.classId.toString(),
-      semester: semester,
-      completion: completion,
-    };
-  });
-}
-/*
-Parses a student's completed courses into a schedule with courses placed in
-whichever semester they were taken. Ignores COOP3945.
+/**
+ * Parses a student's completed courses into a schedule with courses randomly populated into the schedule.
+ * Note: we approximate "when" courses were taken by sorting by prereq order
+ * @param completedCourses the completed courses
  */
-export function parseCompletedCourses(completedCourses: ScheduleCourse[]) {
+export function parseCompletedCourses(
+  completedCourses: ScheduleCourse[]
+): { schedule: DNDSchedule; counter: number } {
   const years: number[] = [];
   const yearMap: { [key: number]: ScheduleYear } = {};
-  const termMap: { [key: string]: SeasonWord } = {
-    "10": "fall",
-    "30": "spring",
-    "40": "summer1",
-    "60": "summer2",
+
+  addBlankScheduleYear(years, yearMap, 2022);
```

Comment:
question: why do we have only add a 2022 blank schedule year? And why specifically 2022? Do we just dump completed courses into a 2022 year or am i misunderstanding this?

Reply:
oh yeah i need to fix this, thanks!

Reply:
just tested it out and it seems like currently the years get overwritten to start at graduating year - current years at neu, and dont' actually use this at all. i'll change it to a constant and leave a comment

Reply:
previously it was getting overwritten in `studentReducer.ts` by `alterScheduleToHaveCorrectYears` but rather than having two tightly-coupled functions, i just pulled out a function. should be good now